DUI3-DX/Converters/Rhino/Speckle.Converters.Rhino7/Geometry/BrepToSpeckleConverter.cs

```diff
@@ -0,0 +1,200 @@
+    // }
+
+    // Vertices, uv curves, 3d curves and surfaces
+    var vertices = target.Vertices.Select(vertex => _pointConverter.RawConvert(vertex.Location)).ToList();
+    var curves3d = target.Curves3D.Select(curve3d => _curveConverter.RawConvert(curve3d)).ToList();
+    var surfaces = target.Surfaces.Select(srf => _surfaceConverter.RawConvert(srf.ToNurbsSurface())).ToList();
+
+    List<ICurve> curves2d;
+    using (_contextStack.Push(Units.None))
+    {
+      // Curves2D are unitless, so we convert them within a new pushed context with None units.
+      curves2d = target.Curves2D.Select(curve2d => _curveConverter.RawConvert(curve2d)).ToList();
+    }
+
+    var speckleBrep = new SOG.Brep
+    {
+      Vertices = vertices,
+      Curve3D = curves3d,
+      Curve2D = curves2d,
+      Surfaces = surfaces,
+      displayValue = displayValue,
+      IsClosed = target.IsSolid,
+      Orientation = (SOG.BrepOrientation)target.SolidOrientation,
+      volume = target.IsSolid ? target.GetVolume() : 0,
+      area = target.GetArea(),
+      bbox = _boxConverter.RawConvert(new RG.Box(target.GetBoundingBox(false))),
+      units = _contextStack.Current.SpeckleUnits
+    };
+
+    // Brep non-geometry types
+    var faces = ConvertBrepFaces(target, speckleBrep);
+    var edges = ConvertBrepEdges(target, speckleBrep);
+    var loops = ConvertBrepLoops(target, speckleBrep);
+    var trims = ConvertBrepTrims(target, speckleBrep);
+
+    return speckleBrep;
+  }
+
+  private static List<SOG.BrepFace> ConvertBrepFaces(RG.Brep brep, SOG.Brep speckleParent) =>
+    brep.Faces
+      .Select(
+        f =>
+          new SOG.BrepFace(
+            speckleParent,
+            f.SurfaceIndex,
+            f.Loops.Select(l => l.LoopIndex).ToList(),
+            f.OuterLoop.LoopIndex,
+            f.OrientationIsReversed
+          )
+      )
+      .ToList();
+
+  private List<SOG.BrepEdge> ConvertBrepEdges(RG.Brep brep, SOG.Brep speckleParent) =>
+    brep.Edges
+      .Select(
+        edge =>
+          new SOG.BrepEdge(
+            speckleParent,
+            edge.EdgeCurveIndex,
+            edge.TrimIndices(),
+            edge.StartVertex?.VertexIndex ?? -1,
+            edge.EndVertex?.VertexIndex ?? -1,
+            edge.ProxyCurveIsReversed,
+            _intervalConverter.RawConvert(edge.Domain)
+          )
+      )
+      .ToList();
+
+  private List<SOG.BrepTrim> ConvertBrepTrims(RG.Brep brep, SOG.Brep speckleParent) =>
+    brep.Trims
+      .Select(trim =>
+      {
+        var t = new SOG.BrepTrim(
+          speckleParent,
+          trim.Edge?.EdgeIndex ?? -1,
+          trim.Face.FaceIndex,
+          trim.Loop.LoopIndex,
+          trim.TrimCurveIndex,
+          (int)trim.IsoStatus,
+          (SOG.BrepTrimType)trim.TrimType,
+          trim.IsReversed(),
+          trim.StartVertex.VertexIndex,
+          trim.EndVertex.VertexIndex
+        )
+        {
+          Domain = _intervalConverter.RawConvert(trim.Domain)
```

Comment:
naming on _internalConverter? Maybe we should have a brep conversion services class? So we kind of outsource the common elements maybe, so long as it doesn't become call-stack depth for no value, though I can see some complexity we might want to be able to test in isolation i.e. I do X with ConvertBrepTrims() and I expect this stuff to happen.

Reply:
Also should go as part of https://spockle.atlassian.net/browse/CNX-9160